catkin_ws/src/planner/src/substates/octagon_task.py

```diff
@@ -33,28 +33,25 @@ def execute(self, ud):
         self.thread_timer = threading.Timer(self.time_limit, self.timer_thread_func)
         self.thread_timer.start()
 
-        # MOVE TO MIDDLE OF POOL DEPTH AND FLAT ORIENTATION
-        self.control.move((None, None, rospy.get_param("down_cam_search_depth")))
         self.control.flatten()
 
-        auv_current_position = (self.state.x, self.state.y)
-
-        octagon_obj = self.mapping.getClosestObject(
-            "Octagon Table", (auv_current_position[0], auv_current_position[1])
-        )
+        octagon_obj = self.mapping.getClosestObject((self.state.x, self.state.y), cls="Octagon Table") #Assume that we can see the octagon + reasonably close
 
         if octagon_obj is None:
             print("No octagon in object map! Failed.")
             return "failure"
 
         if self.timeout_occurred:
             return "timeout"
+        
         print("Moving to the center of the octagon.")
         self.control.move(
             (octagon_obj[1], octagon_obj[2], rospy.get_param("down_cam_search_depth")),
             face_destination=True,
         )
```

Comment:
It looks really good. The only problem is that it doesn't wait for the object map to improve. Rn, it sets the pid after the first detection. But the first detection does not always translate to a good object map. So, you can either wait a bit for the object map to update and correct itself or update the pid as the object map improves (with some loop)

Reply:
After testing in the sim, it sometimes kills the thrusters too early and it ends up hitting the sides of the octagon instead of being in the middle.

Reply:
Made a loop that gets the location of the octagon as the AUV moves towards it then breaks when the AUV is close enough to the octagon that we think it can accurately detect its location. Currently that threshold is set to 3. Also fixed typos in object_map.py